src/Evolution/Systems/GeneralizedHarmonic/Constraints.cpp

```diff
@@ -272,6 +273,650 @@ void two_index_constraint_add_term_11_of_11(
     }
   }
 }
+
+// Functions to compute generalized-harmonic F constraint, where
+// function arguments are in the order that each quantity first
+// appears in the corresponding term in Eq. (43) of
+// https://arXiv.org/abs/gr-qc/0512093v3
+template <size_t SpatialDim, typename Frame, typename DataType>
+void f_constraint_add_term_1_of_25(
+    const gsl::not_null<tnsr::a<DataType, SpatialDim, Frame>*> constraint,
+    const tnsr::a<DataType, SpatialDim, Frame>& spacetime_normal_one_form,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_normal_vector,
+    const tnsr::AA<DataType, SpatialDim, Frame>& inverse_spacetime_metric,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& d_pi) noexcept {
+  for (size_t a = 0; a < SpatialDim + 1; ++a) {
+    for (size_t b = 0; b < SpatialDim + 1; ++b) {
+      for (size_t c = 0; c < SpatialDim + 1; ++c) {
+        if (a > 0) {
+          constraint->get(a) +=
+              0.5 * inverse_spacetime_metric.get(b, c) * d_pi.get(a - 1, b, c);
```

Comment:
I think the current version is easier to read when comparing to the paper where the equation came from, so I would like to keep this line as is for this PR. For an issue after this PR, I will profile whether precomputing makes sense here.

Reply:
I don't understand. In the paper the equations are using g^i_a, so it would be easier to compare if you used it.

Reply:
I see what you're saying about precomputing being easier to read. But after thinking about this more, I still want to postpone trying to precompute things by making an issue to profile this term to see whether precomputing makes sense, rather than making the change in this PR.

It's not obvious to me that the current version has no performance advantage: by analytically applying the Kronecker delta in the projection operator, I avoid some multiplications by zero in the summations, and I also have fewer allocations.